lib/stronger_parameters/constraints/decimal_constraint.rb

```diff
@@ -0,0 +1,28 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  class DecimalConstraint < Constraint
+    attr_reader :precision, :scale
+
+    def initialize(precision, scale)
+      @precision = precision
+      @scale = scale
+    end
+
+    def value(v)
+      if scale > 0
+        if v.is_a?(Float) && (v.abs.to_s.size - 1) <= precision
+          return value(v.to_s)
+        elsif v.is_a?(String) && v =~ /\A-?\d{1,#{precision - scale}}\.?\d{#{scale}}\Z/
+          return v.to_d
+        end
+      elsif (v.is_a?(Fixnum) || v.is_a?(Bignum)) && (v.abs.to_s.size <= precision)
+        return v.to_d
+      elsif v.is_a?(String) && v =~ /\A-?\d{1,#{precision - scale}}\Z/
```

Comment:
I tried to include a broad spectrum in my tests, I'm not sure if there could be other input possibilities. What I'm currently thinking though is that when you for instance have a 8/2 precision/scale, it isn't possible to accept integers. As it always is a valid decimal. What do you think?